src/contract.ts

```
@@ -0,0 +1,381 @@
+      simulate,
+      instruction,
+      signers
+    );
+
+    let result: ethers.utils.Result | null = null;
+
+    if (fragment.outputs?.length) {
+      if (!encoded) {
+        throw new Error('return data not set');
+      }
+
+      result = this.abi.decodeFunctionResult(fragment, encoded);
+      if (fragment.outputs.length === 1) {
+        result = result[0];
+      }
+    }
+
+    const events = this.parseLogsEvents(logs);
+
+    return { result, logs, computeUnitsUsed, events };
+  }
+
+  /**
+   *
+   * @param test
+   * @param upto
+   * @returns
+   */
+  async contractStorage(test: Program, upto: number): Promise<Buffer> {
+    const accountInfo = await test.connection.getAccountInfo(
+      this.contractStorageAccount.publicKey
+    );
+
+    return accountInfo!.data;
+  }
+
+  /**
+   *
+   * @returns
+   */
+  getStorageKeyPair(): Keypair {
+    return this.contractStorageAccount;
+  }
+
+  /**
+   * Invokes the given callback every time the given event is emitted.
+   *
+   * @param callback  The function to invoke whenever the event is emitted from
+   *                  program logs.
+   */
+  public addEventListener(callback: EventCallback): number {
+    return this.program.addEventListener(this.abi, callback);
+  }
+
+  /**
+   * Unsubscribes from the given eventName.
+   */
+  public async removeEventListener(listener: number): Promise<void> {
+    return await this.program.removeEventListener(listener);
+  }
+
+  /**
+   *
+   * @param logs
+   * @returns
+   */
+  public parseLogsEvents(logs: string[]): ethers.utils.LogDescription[] {
+    const events: ethers.utils.LogDescription[] = [];
+
+    for (const log of logs) {
+      const eventData = parseLogTopic(log);
+      if (eventData) {
+        let event: ethers.utils.LogDescription | null = null;
+        try {
+          event = this.abi.parseLog(eventData);
+        } catch (e) {
+          // console.log(e);
```

What can cause a silent error here?

---

Function cannot fail unless buggy, no need for try/catch

---

Fixed.